### PR TITLE
Added new datetime functions and aliases to PPL

### PR DESCRIPTION
**Pull request by matthewryanwells**
### Description
This PR adds multiple datetime functions and aliases to PPL.

The following aliases are added:
| Alias | Original Function |
| --- | --- |
| day_of_week | dayofweek |
| day_of_month | day |
| hour_of_day | hour |
| minute_of_hour | minute |
| second_of_minute | second |
| day_of_year | dayofyear |
| month_of_year | month |
| week_of_year | week |

The following functions are added:
- minute_of_day
- time_format

### Issues Resolved
No related issue
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/user/ppl/functions/datetime.rst [hunk 1]
```diff
@@ -688,6 +688,31 @@ Example::
     +----------------------------------+
 
 
+DAY_OF_MONTH
+----------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:692]:
```suggestion
DAY_OF_MONTH
------------
```

File: docs/user/ppl/functions/datetime.rst [hunk 2]
```diff
@@ -711,6 +738,30 @@ Example::
     +---------------------------------+
 
 
+DAY_OF_WEEK
+---------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:742]:
```suggestion
DAY_OF_WEEK
-----------
```

File: docs/user/ppl/functions/datetime.rst [hunk 3]
```diff
@@ -735,6 +788,31 @@ Example::
     +---------------------------------+
 
 
+DAY_OF_YEAR
+---------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:792]:
```suggestion
DAY_OF_YEAR
-----------
```

File: docs/user/ppl/functions/datetime.rst [hunk 4]
```diff
@@ -817,6 +897,31 @@ Example::
     +--------------------------+
 
 
+HOUR_OF_DAY
+----
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:901]:
```suggestion
HOUR_OF_DAY
-----------
```

Comment (Yury-Fridlyand) [reply docs/user/ppl/functions/datetime.rst:901]: Fix this everywhere please

File: ppl/src/main/antlr/OpenSearchPPLLexer.g4 [hunk 5]
```diff
@@ -95,11 +95,18 @@ INTERVAL:                           'INTERVAL';
 MICROSECOND:                        'MICROSECOND';
 MILLISECOND:                        'MILLISECOND';
 SECOND:                             'SECOND';
+SECOND_OF_MINUTE:                   'SECOND_OF_MINUTE';
```
Comment (Yury-Fridlyand) [inline ppl/src/main/antlr/OpenSearchPPLLexer.g4:98]: Can you groom this list to be in alphabetical order?

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 6]
```diff
@@ -521,32 +521,41 @@ dateTimeFunctionName
     | DATE
     | DATE_ADD
     | DATE_FORMAT
+    | TIME_FORMAT
```
Comment (Yury-Fridlyand) [inline ppl/src/main/antlr/OpenSearchPPLParser.g4:524]: Please keep it in alphabetical order

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

File: docs/user/ppl/functions/datetime.rst (hunk 1 above)
Comment (matthewryanwells) [reply docs/user/ppl/functions/datetime.rst]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: docs/user/ppl/functions/datetime.rst (hunk 2 above)
Comment (matthewryanwells) [reply docs/user/ppl/functions/datetime.rst]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: docs/user/ppl/functions/datetime.rst (hunk 3 above)
Comment (matthewryanwells) [reply docs/user/ppl/functions/datetime.rst]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: docs/user/ppl/functions/datetime.rst (hunk 4 above)
Comment (matthewryanwells) [reply docs/user/ppl/functions/datetime.rst]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: ppl/src/main/antlr/OpenSearchPPLLexer.g4 (hunk 5 above)
Comment (matthewryanwells) [reply ppl/src/main/antlr/OpenSearchPPLLexer.g4]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 (hunk 6 above)
Comment (matthewryanwells) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4]: fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/7b927df652fd09ef0e74315eb6a2abfbffe1f8f3

File: ppl/src/main/antlr/OpenSearchPPLLexer.g4 [hunk 7]
```diff
@@ -247,11 +254,14 @@ ADDTIME:                            'ADDTIME';
 DATE:                               'DATE';
 DATE_ADD:                           'DATE_ADD';
 DATE_FORMAT:                        'DATE_FORMAT';
+TIME_FORMAT:                        'TIME_FORMAT';
```
Comment (Yury-Fridlyand) [inline ppl/src/main/antlr/OpenSearchPPLLexer.g4:257]: a T spy in D party

**Review by Yury-Fridlyand — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

File: docs/user/ppl/functions/datetime.rst [hunk 8]
```diff
@@ -627,7 +627,7 @@ Argument type: STRING/DATE/DATETIME/TIMESTAMP
 
 Return type: INTEGER
 
-Synonyms: `DAYOFMONTH`_
+Synonyms: `DAYOFMONTH`_, `DAY_OF_MONTH`_
```
Comment (acarbonetto) [inline docs/user/ppl/functions/datetime.rst:630]: `DAY` as well?

**Review by acarbonetto — APPROVED** [review]

File: ppl/src/main/antlr/OpenSearchPPLLexer.g4 (hunk 7 above)
Comment (matthewryanwells) [reply ppl/src/main/antlr/OpenSearchPPLLexer.g4]: Good catch, updated in https://github.com/Bit-Quill/opensearch-project-sql/pull/281/commits/6fa69170702814d04b302773f51e016cf64f7674

**Pull request merged**